drainnodes.sh

```diff
@@ -0,0 +1,209 @@
+    echo "Options:";
+    echo "  -l  --selector  selectors (label query) to filter nodes on";
+    echo "  -c  --count     count of non-running pods (completed/error) in the cluster before starting draining process (default 2)";
+    echo "      --dry-run   simulate nodes drain";
+    echo "  -y  --yes       run non interractively";
+    echo "      --no-color  remove the additional color from the output";
+}
+
+
+while [[ $# -gt 0 ]]
+do
+key="$1"
+
+case $key in
+    -l|--selector)
+    LABELS+=("-l $2")
+    shift
+    shift
+    ;;
+    -c|--count)
+    POD_COUNT="$2"
+    shift
+    shift
+    ;;
+    --dry-run)
+    INFO_ONLY=true
+    shift
+    ;;
+    --no-color)
+    NO_COLOR=true
+    shift
+    ;;
+    -y|--yes)
+    NON_INTERRACTIVE=true
+    shift
+    shift
+    ;;
+    -h|--help)
+    output_help
+    exit 0
+    shift
+    ;;
+    *)
+    NAMES+=("$1")
+    shift
+    ;;
+esac
+done
+
+# ----------------
+#  ECHO COLORING
+# ----------------
+
+# $1 content to echo
+# $2 color
+function c_echo {
+    if [ "$NO_COLOR" = true ] ; then
+        echo $1
+    fi
+    RED='\033[0;31m'
+    GREEN='\033[0;32m'
+    YELLOW='\033[1;33m'
+    CYAN='\033[0;36m'
+    NC='\033[0m' # No Color
+    case $2 in
+        red)
+            printf "${RED}%b${NC}\n" "${1}"
+        ;;
+        green)
+            printf "${GREEN}%b${NC}\n" "${1}"
+        ;;
+        yellow)
+            printf "${YELLOW}%b${NC}\n" "${1}"
+        ;;
+        cyan)
+            printf "${CYAN}%b${NC}\n" "${1}"
+        ;;
+        *)
+            printf "%b\n" "${1}"
+    esac
+}
+
+# ----------------
+#  HELPER FUNCTIONS
+# ----------------
+
+function are_pods_not_running() {
+    local counter
+    counter=0
+    while read -r w1 w2 w3 w4 w5 ; do
+        if [ $w4 != "Running" ] && [ $w4 != "STATUS" ]; then
+            let counter=counter+1
+        fi
+    done <<< "$($COUNT_PODS)"
+    if [ "$counter" -gt "$1" ] ; then
+        true
+    else
+        false
+    fi
+}
+
+# $1 nodeName
+# $2 additional flag
+function drain_node() {
+    local output
+    c_echo ""
+    c_echo " $ kubectl drain --ignore-daemonsets --delete-local-data $2 $1"
+    kubectl drain --ignore-daemonsets --delete-local-data $2 $1
+}
+
+function wait_for_pods_to_migrate() {
+    c_echo "Waiting for all (no less than $POD_COUNT) pods to start running again"
+    while are_pods_not_running $POD_COUNT ; do
+        printf "."
+        sleep 5
+    done
+    c_echo "   done waiting"
+}
+
+# $1 nodeName
+function echo_resource_usage() {
+    c_echo "  resource usage for: $1  " "cyan"
+    c_echo "  ------------------------"
+    c_echo "$(kubectl describe node $1 | grep Allocated -A 5 |
+            grep -ve Event -ve Allocated -ve percent -ve --)"
+    c_echo ""
+}
+
+if are_pods_not_running $POD_COUNT ; then
+    c_echo "Not enough pods are running. Adjust with -c option, and/or" "red"
+    c_echo "  check which pods are running/not:" "red"
+    c_echo "    $COUNT_PODS | grep -ve Running"
```

Comment:
Could be clarified ? 
Suggestion:
\<N> pods are already unhealthy (\<M> pods ignored): the drain operation will not start (because the script will not be able to detect if the migration is complete). Use the "-c POD_COUNT" option to accept POD_COUNT unhealthy pods.